### PR TITLE
Modified libpng header include path

### PR DESCRIPTION
**Pull request by Vixian**
`/usr/include/png.h` is the standard include directory for `png.h`. It is a symlink to `/usr/include/libpng<version>/png.h`

While the OpenCV version before this change does successfully find *libpng* on Ubuntu, because installing *libpng-dev* also creates `/usr/include/libpng/png.h`. It doesn't do so on Fedora, where there is no `libpng` directory only `libpng<version>` and a directoryless simlink. To be precise FindPNG finds the right include directory, but the log outputs it didn't.

This is a more upstream friendly way of linking png.h.
Also, this allows to include png.h using `#include <png.h>` regardless if it is build from source or found in the system. Which is the upstream way of including `png.h` as can be seen here [Example](http://zarb.org/~gc/html/libpng.html) and in more examples across the internet.

### CODE REVIEW

**Comment by alalek** [comment]
if there is no "libpng/png.h" file, then there is the branch that check "png.h".
I have Fedora 21 and there are no issues with using of system libpng. CMake output:
```
-- Found PNG: /lib64/libpng.so (found version "1.6.10") 
-- Looking for /usr/include/libpng/png.h
-- Looking for /usr/include/libpng/png.h - not found
...
--   Media I/O: 
--     ZLib:                        /lib64/libz.so (ver 1.2.8)
--     JPEG:                        /lib64/libjpeg.so (ver )
--     WEBP:                        build (ver 0.3.1)
--     PNG:                         /lib64/libpng.so (ver 1.6.10)
--     TIFF:                        /lib64/libtiff.so (ver 42 - 4.0.3)
--     JPEG 2000:                   build (ver 1.900.1)
--     OpenEXR:                     build (ver 1.7.1)
--     GDAL:                        NO
```
CMake cache contents about PNG:
```
//Path to a library.
PNG_LIBRARY_RELEASE:FILEPATH=/lib64/libpng.so

//Path to a file.
PNG_PNG_INCLUDE_DIR:PATH=/usr/include
```

What is the problem?

**Comment by Vixian** [comment]
As I said, yes, FindPNG finds the correct directory `/usr/include`. I just pointed out that `/usr/include/libpng/png.h` isn't the standard `png.h` directory and from the log output as it is, I came to conclusion that `png.h` is missing. Wouldn't you? 
But it did find it, and only way to figure that out right now is to look at the cache as you just did.

The linking works, I just made the log more telling, and the inclusion simpler.
After the changes the log will say:
```
-- Looking for /usr/include/png.h
-- Looking for /usr/include/png.h - found
```
on both Fedora and Ubuntu

File: cmake/OpenCVFindLibsGrfmt.cmake [hunk 1]
```diff
@@ -157,12 +157,8 @@ if(WITH_PNG)
...
-      check_include_file("${PNG_PNG_INCLUDE_DIR}/libpng/png.h" HAVE_LIBPNG_PNG_H)
-      if(HAVE_LIBPNG_PNG_H)
-        ocv_parse_header("${PNG_PNG_INCLUDE_DIR}/libpng/png.h" PNG_VERSION_LINES PNG_LIBPNG_VER_MAJOR PNG_LIBPNG_VER_MINOR PNG_LIBPNG_VER_RELEASE)
-      else()
-        ocv_parse_header("${PNG_PNG_INCLUDE_DIR}/png.h" PNG_VERSION_LINES PNG_LIBPNG_VER_MAJOR PNG_LIBPNG_VER_MINOR PNG_LIBPNG_VER_RELEASE)
-      endif()
+      check_include_file("${PNG_PNG_INCLUDE_DIR}/png.h" HAVE_LIBPNG_PNG_H)
+      ocv_parse_header("${PNG_PNG_INCLUDE_DIR}/png.h" PNG_VERSION_LINES PNG_LIBPNG_VER_MAJOR PNG_LIBPNG_VER_MINOR PNG_LIBPNG_VER_RELEASE)
     endif()
   endif()
```
Comment (Vixian) [inline cmake/OpenCVFindLibsGrfmt.cmake]: I just noticed there's also a redundant if branch now, so I removed it.

**Comment by Vixian** [comment]
I just understood the code properly, sorry, it is indeed correct as is, I just didn't find the message:
```
-- Looking for /usr/include/libpng/png.h
-- Looking for /usr/include/libpng/png.h - not found
```
...meaningful but now I understand why it's there.

**Pull request closed without merging**